### ISSUE TITLE
Improve 'Status & visibility' panel in the inspector sidebar

### ISSUE THREAD

**Issue by noisysocks** [WordPress/gutenberg#39082]
Let's improve the _Status & visibliity_ panel that appears in the document settings sidebar.

Resources:

- [Blog post](https://shaunandrews.com/2021/09/the-wordpress-editor-document-status-and-visibility/) by @shaunandrews
- [Design exploration](https://github.com/WordPress/gutenberg/issues/39082#issuecomment-1073974215) by @javierarce

![Content](https://user-images.githubusercontent.com/612155/177481254-742bdd00-5c8b-4786-9bf3-81f75edd9483.png)

## Changes required

[Author popover](https://user-images.githubusercontent.com/4933/159279818-e427abac-672c-4fbb-8fcb-d4e01893d779.png) | [Popovers](https://user-images.githubusercontent.com/4933/159278015-2db02481-aa3b-4d8a-874c-164101c5d688.png)

- [x] https://github.com/WordPress/gutenberg/pull/40530
- [x] https://github.com/WordPress/gutenberg/pull/41097
- [ ] https://github.com/WordPress/gutenberg/pull/41430
- [x] https://github.com/WordPress/gutenberg/pull/41925
- [x] https://github.com/WordPress/gutenberg/pull/42033

[Summary panel](https://user-images.githubusercontent.com/4933/159283464-29314f93-d934-420e-aa32-50d30aebce6c.png) | [Secondary options](https://user-images.githubusercontent.com/4933/159279715-18cb656b-9eaa-4cbe-ab0f-bd099b0f23d6.png)

- [ ] https://github.com/WordPress/gutenberg/pull/42175

[Editable feature image](https://user-images.githubusercontent.com/4933/159277376-af5521bf-9e70-49f6-8c37-68d0b1c6f743.png) | [Inline editing](https://user-images.githubusercontent.com/4933/159277543-a9663710-97b4-42af-a893-ab39c60e72ba.png)

- [ ] https://github.com/WordPress/gutenberg/pull/42352

**Comment by noisysocks** [comment]
Hi @javierarce! Thanks for volunteering to help with this. How do the designs in @shaunandrews's post look to you? Let's get them finalised before I begin development.

I also had a couple of questions:

- Do you have a link to the mockups in Figma?
- Where should the _Stick to the top of the blog_ and _Pending review_ checkboxes go? I don't see them in the mockups.

**Comment by javierarce** [comment]
> How do the designs in @shaunandrews's post look to you? Let's get them finalised before I begin development.

They look good, but I want to review them again since I haven't looked at them in a while.

> Do you have a link to the mockups in Figma?

Here's the [original file](https://www.figma.com/file/WBBq8uhaKydqVunjCa9T77/?node-id=104%3A6689). Since it contains lots of explorations for other elements of the sidebar, I'm going to create a new file to focus on this project.

> Where should the Stick to the top of the blog and Pending review checkboxes go? I don't see them in the mockups.

The "Stick to the top of blog list" is accessible through the pin icon.

![image](https://user-images.githubusercontent.com/4933/155710174-8709c768-f4fc-412e-9b80-d330f8c5ccf1.png)

I'm not sure where he placed the review checkbox on that exploration, but on a different one, he kept the checkboxes in the sidebar.

![image](https://user-images.githubusercontent.com/4933/155710274-c32e3537-2bd7-406e-b81e-0f151f2e77c4.png)

**Comment by noisysocks** [comment]
Thanks @javierarce! I'll wait for your new Figma before starting dev 😀

**Comment by tellthemachines** [comment]
Regarding the visibility radio buttons, it would be good to bear in mind the a11y issue described in #15328. The tl;dr is we should avoid popping up a dialog on switching the visibility to private. The classic editor gets around this by silently publishing the post (if it's a draft). An alternative could be to add a warning inline or underneath the private option, to make it clear that the switch to private will auto-publish the post.

**Comment by javierarce** [comment]
Thanks, @tellthemachines, I'll have that in mind.

**Comment by javierarce** [comment]
Here's the [Figma file](https://www.figma.com/file/8vEDwRwmzWhxvcvhc6yaQk/Status-%26-visibility?node-id=1%3A2) for my exploration. It includes [an interactive prototype](https://www.figma.com/proto/8vEDwRwmzWhxvcvhc6yaQk/Status-%26-visibility?page-id=1%3A2&node-id=1%3A2&scaling=min-zoom&starting-point-node-id=343%3A2605&show-proto-sidebar=1) that explains the whole concept and gives some context through the lens of two other proposals. I encourage anyone to have a look before reading the rest of the issue.

The main goal of my exploration is to improve the overall legibility of the sidebar. To accomplish that, I propose reorganizing its sections (so that the most important items appear on top) and grouping the settings (so they are easier to find and manage). By applying those changes, we'll also reduce the height of the toolbar.

## Structure

The sidebar I propose is divided in three main sections: 

- Summary area
- Settings area
- Organization & discussion area

![image](https://user-images.githubusercontent.com/4933/159283464-29314f93-d934-420e-aa32-50d30aebce6c.png)

The summary area that sits on top of the sidebar acts as a card for the post and contains a preview of the featured image, the title, and the excerpt, plus the user (or users) that collaborated on the post.

Here's what this area would look like before and after the user has filled up the content:

![image](https://user-images.githubusercontent.com/4933/159277376-af5521bf-9e70-49f6-8c37-68d0b1c6f743.png)

Promoting the featured image and the excerpt to a top position encourages good online practices: filling up that meta-information will potentially increase the reach of the posts when shared online.

If the empty placeholder of the featured image is too prominent, we could find other ways to make it less noticeable but equally accessible. Here are a couple of ideas:

![image](https://user-images.githubusercontent.com/4933/159277410-a1624658-bad0-4a52-b844-ab1ef224c6ef.png)

Secondary options like the one to stick the post to the top of a blog, accessing the revision history, or removing the post could be accessed through the ellipsis menu of the section.

![image](https://user-images.githubusercontent.com/4933/159279715-18cb656b-9eaa-4cbe-ab0f-bd099b0f23d6.png)

And editing the title and the excerpt could use the inline editing proposal from Shaun's exploration, which is compact and easy to use.

![image](https://user-images.githubusercontent.com/4933/159277543-a9663710-97b4-42af-a893-ab39c60e72ba.png)

Modifying the other fields will be done through popovers. 

Here's my proposal for the Author popover.

![image](https://user-images.githubusercontent.com/4933/159279818-e427abac-672c-4fbb-8fcb-d4e01893d779.png)

As you can see in the image above, each item contains:  

- An avatar
- The display name
- A label with secondary information

We could also indicate (using an icon or a text string) which user is the owner or the original creator of the post. 

Also, instead of showing the email address, we could show the user's role.

![image](https://user-images.githubusercontent.com/4933/159277728-ebc6d6cb-b61f-47ba-a755-0e2da79b8445.png)


### Adding users  

The input field on the top of the popover allows searching and adding users to the list. If we showed the email address, we could allow searching for both the display name and the email.

![image](https://user-images.githubusercontent.com/4933/159279947-eca6f3df-9ad7-45d7-bb55-cbb7671105f7.png)

## Removing users

 When it comes to removing users, there are two rules: 
 
- We don't allow removing the last author. Instead, when there's just one user on the list, we disable the remove option.
- We allow removing the owner of the post, but we show a message to indicate that they were removed. This could be useful to locate the original author.

![image](https://user-images.githubusercontent.com/4933/159277911-8307a371-d95c-4d8d-b17f-79c70deab477.png)

### Publish settings

  Right below the summary area, there could be a section that groups all the settings related to how and when the post is presented to its readers.

That section would contain settings for:

- Visibility 
- Scheduling 
- URL 
- Template

![image](https://user-images.githubusercontent.com/4933/159277971-13c6b37d-a1fb-4c24-80c6-6bf1741288c6.png)

As with the post authors, all those settings would be accessible through different popovers, which help to reduce the visual clutter on the sidebar and allow users to focus on one main task.

Secondary actions are accessed through a button or link in the header of each popover.

![image](https://user-images.githubusercontent.com/4933/159278015-2db02481-aa3b-4d8a-874c-164101c5d688.png)

I won't cover each of them in this issue, but you can see how the new ones could work in the following Figma prototypes plus some iterations for each of them:

- [Date popover](https://www.figma.com/proto/8vEDwRwmzWhxvcvhc6yaQk/Status-%26-visibility?page-id=1%3A2&node-id=404%3A15161&viewport=352%2C48%2C0.18&scaling=min-zoom&starting-point-node-id=404%3A15161&show-proto-sidebar=1)
- [URL popover](https://www.figma.com/proto/8vEDwRwmzWhxvcvhc6yaQk/Status-%26-visibility?page-id=1%3A2&node-id=405%3A26933&viewport=352%2C48%2C0.18&scaling=min-zoom&starting-point-node-id=405%3A26933&show-proto-sidebar=1)
- [Template popover](https://www.figma.com/proto/8vEDwRwmzWhxvcvhc6yaQk/Status-%26-visibility?page-id=1%3A2&node-id=412%3A14269&viewport=352%2C48%2C0.18&scaling=min-zoom&starting-point-node-id=412%3A14269&show-proto-sidebar=1)
- [Visibility popover](https://www.figma.com/proto/8vEDwRwmzWhxvcvhc6yaQk/Status-%26-visibility?page-id=1%3A2&node-id=769%3A19985&viewport=352%2C48%2C0.18&scaling=min-zoom&starting-point-node-id=343%3A2605)

### Iterations

To finish this comment, I'll show you a couple of simple iterations on the sidebar using the same ideas I've already presented. The one on the left is a little more compact than the sidebar I just showed you. The one on the right demotes the excerpt field.

![image](https://user-images.githubusercontent.com/4933/159282796-75b14350-57ed-4a4a-b98a-248770e9833b.png)

And here's another one that includes the creation and update dates:

![image](https://user-images.githubusercontent.com/4933/161018677-f8d444b2-f8fd-40e0-ab82-00a8d681a6bc.png)

**Comment by noisysocks** [comment]
Looks great @javierarce!

I prefer the variation where everything is underneath _Summary_ and not split into _Summary_ and _Publish settings_. It means that we don't have to try and label the settings which can be unintuitive. For instance, why is _Template_ a "publish setting"?

The mockups show the ability to have multiple authors, but I don't think that you can do this in WordPress without using a plugin? It's a cool feature, but would be a significant undertaking. Best to explore that separately from this project.

**Comment by noisysocks** [comment]
Updated _Changes required_ in the description to reflect how I'd break down implementing the updated design. I probably won't be able to start on development until I'm back from vacation in mid April, though 😅

**Comment by javierarce** [comment]
> I prefer the variation where everything is underneath Summary and not split into Summary and Publish settings. It means that we don't have to try and label the settings which can be unintuitive. For instance, why is Template a "publish setting"?

Maybe we could find a better name for the section. My only worry is that we end up with more than 5 or 6 options in there (the idea for the ellipsis menu for the "publish settings" would be to configure the items that appear there).

> The mockups show the ability to have multiple authors, but I don't think that you can do this in WordPress without using a plugin? It's a cool feature, but would be a significant undertaking. Best to explore that separately from this project.

I'm totally OK to leave it out of this project, it was more of an exploration for the future. I'll add a variation that only shows one user and can be implemented easily.

**Comment by jasmussen** [comment]
This is looking good! This part alone feels potent:

<img width="670" alt="top of the status and visibility panel, showing featured image, title, excerpt and author" src="https://user-images.githubusercontent.com/1204802/159647089-e36decf6-530c-48ed-800b-149d819dd2af.png">

This segment reads almost like a "colophon", an inscription of publication facts that you'll usually see at the end of a book. It's reads like a single piece of prose, yet it integrates title, featured image, excerpt and author in an organic and intuitive structure.

Even if this one piece is the only thing we implement as a starting point, it'll be an excellent improvement.

**Comment by ZebulanStanphill** [comment]
I really like these mockups from a visual and organization perspective, but I question the accessibility here. There seems to be a lot of buttons that are labeled with the value of a setting, rather than an actual description of what the button does. Granted, the current design already has this problem, but these mockups make the issue even more prevalent by extending it to stuff like the author and summary fields.

#470 is one of the oldest issues in this repository, and unfortunately, these mockups seem to worsen the described problem.

**Comment by ZebulanStanphill** [comment]
I really like these mockups from a visual and organization perspective, but I question the accessibility here. There seems to be a lot of buttons that are labeled with the value of a setting, rather than an actual description of what the button does. Granted, the current design already has this problem, but these mockups make the issue even more prevalent by extending it to stuff like the author and summary fields.

#470 is one of the oldest issues in this repository, and unfortunately, these mockups seem to worsen the described problem.

And I should note that I'm not just talking about screen readers. In fact, I'm mostly talking about visual usability. It genuinely is confusing to have buttons not be labeled according to their actions. If I see blue text with the name of an author, I might think that it's a link to their profile, rather than a way to edit the author list.

**Comment by jasmussen** [comment]
@noisysocks What if we start with just the colophon part as [outlined here](https://github.com/WordPress/gutenberg/issues/39082#issuecomment-1076019494)? I'd be super happy to pair on it, do the CSS/style work.

**Comment by ntsekouras** [comment]
>@noisysocks What if we start with just the colophon part as https://github.com/WordPress/gutenberg/issues/39082#issuecomment-1076019494? I'd be super happy to pair on it, do the CSS/style work.

FYI I'll start a first version of this with @jasmussen' s support 😄

**Comment by jasmussen** [comment]
This week, @ntsekouras and I tried a V1 of the Summary panel to see if including just [the colophon part](https://github.com/WordPress/gutenberg/issues/39082#issuecomment-1076019494) could work. A few issues emerged, so we'll be reverting it for now. What we learned from the effort can help inform the future direction of this ticket. 

Mainly, the omission of the ellipsis control for managing metadata came with a few downsides:

* Instead of becoming mainly a _summary_, it fully absorbed Excerpt and Featured Image panels, making it less clear how to add/edit those.
* In giving prime real estate to the empty placeholders, it gave prominence to the excerpt and featured image features that is not always deserved: they are crucial for posts that need them, but not all posts do. 

Both of these issues feel partially solved already through the ellipsis functionality outlined on this ticket, suggesting the V1 of this feature should include that functionality.

One take-away worth revisiting is the flow of _how_ excerpts and images are added: perhaps instead of being placeholders to fill out, the summary can be mostly a static representation of only that which is added. The act of adding these can then remain a separate part of the summary. Something to explore!

The effort did validate the clear context the metadata gives when filled out, showing an elegant summary of the document. In fact, when this returns, we should consider adding and integrating this summary also in the pre-publish flow.

**Comment by noisysocks** [comment]
Thanks for the exploration @ntsekouras and @jasmussen! I'm back at work now so will get going on this.

> Instead of becoming mainly a summary, it fully absorbed Excerpt and Featured Image panels, making it less clear how to add/edit those.

I think it's pretty clear how to _add_ a title, excerpt, and featured image because we show a placeholder.

<img width="290" alt="Screen Shot 2022-04-19 at 14 33 35" src="https://user-images.githubusercontent.com/612155/163920784-6f782aed-8bf4-4d6c-87cc-01f5a205d439.png">

But you're right that, when these fields have content, it's not very clear that these fields are editable.

<img width="291" alt="Screen Shot 2022-04-19 at 14 34 21" src="https://user-images.githubusercontent.com/612155/163920849-3bc1dab0-2389-444a-bc43-f5ce8d66b1a8.png">

Maybe we can go back to showing an outline as in @shaunandrews's proposal or use a pencil icon as in David's (sorry, don't know his handle) proposal.

<img width="294" alt="Screen Shot 2022-04-19 at 14 35 49" src="https://user-images.githubusercontent.com/612155/163920993-47028c27-2dcb-4f99-9b5c-d2f79bde3b64.png">
<img width="293" alt="Screen Shot 2022-04-19 at 14 36 09" src="https://user-images.githubusercontent.com/612155/163921030-c2e0e8b8-8be8-4d30-91cd-7b8a2d90c09c.png">

I'll leave it up to @javierarce to iterate here 😀

> In giving prime real estate to the empty placeholders, it gave prominence to the excerpt and featured image features that is not always deserved: they are crucial for posts that need them, but not all posts do.

Well put. It's why I prefer the variation that makes _Set featured image_ appear less prominent and keeps _Excerpt_ in its own dedicated panel.

**Comment by jasmussen** [comment]
One idea I think could be interesting to explore:

* Keeping the summary panel truly _inert_ — literally a summary.
* Don't show any placeholders at all, see how it looks if we summarize only that which has been filled out. If you haven't filled out an excerpt or a featured image, you'd get only the Title and the Author. 
* This summary could play a key role in the pre/post-publish flow.
* Excerpt and Featured Image panels _remain_, either as bespoke collapsible panels as they are today, or as sections that can be added using an ellipsis menu flow.
  * Doing so would allow us to start with the Summary and postpone any ellipsis mechanisms, as a followup step.

The above would not preclude future explorations in terms of retirement or otherwise, but based on our experiment, it seems like the most pragmatic way to start. What do you think?

**Comment by javierarce** [comment]
@noisysocks:

> Maybe we can go back to showing an outline as in @shaunandrews's proposal or use a pencil icon as in David's (sorry, don't know his handle) proposal.

Those two ideas are also present in my prototype, so I'd say let's go for it :)

@jasmussen:

> The above would not preclude future explorations in terms of retirement or otherwise, but based on our experiment, it seems like the most pragmatic way to start. What do you think?

The problem I see with this approach is that, even if this could be useful (which I think it could be), we would be adding more elements to the interface instead of removing stuff. For me, this idea could be a good middle point only if we end up implementing the editable version.

**Comment by jasmussen** [comment]
To expand a bit, I mean for that suggested approach to be a stepping stone, not the final destination. It may be easier to implement in steps, and doing so may unearth challenges along the way. The ellipsis to add/subtract properties seems promising enough!

**Comment by javierarce** [comment]
> To expand a bit, I mean for that suggested approach to be a stepping stone, not the final destination.

Oh, perfect then!

**Comment by noisysocks** [comment]
> * This summary could play a key role in the pre/post-publish flow.

Could you elaborate on this a bit? Would we show the summary in the pre-publish panel as well?

**Comment by jasmussen** [comment]
> Could you elaborate on this a bit? Would we show the summary in the pre-publish panel as well?

Sure, and this wouldn't necessarily have to be part of any near-future PRs, it's just a general idea. Essentially, the following area of the pre-publish dialog (press Publish) is currently the closest thing we have to a Summary panel today:
![prepublish](https://user-images.githubusercontent.com/1204802/164164951-e9560c90-4219-485e-b0fc-3e23ceebf537.gif)

Designwise, it could make good sense to share DNA between the panels here. It needs a little design exploration, though.

**Comment by jameskoster** [comment]
The idea of moving the "Switch to draft" button to the Inspector came up in https://github.com/WordPress/gutenberg/issues/40891. Any ideas where that might fit in @javierarce?

**Comment by javierarce** [comment]
@jameskoster I think we could put it inside the Visibility dropdown, since it controls who can see the post. We could have: 

- Draft: Visible to you
- Public: Visible to everybody
- Private: Only visible to site admins and editors
- Password Only those with the password can view this post

**Comment by jameskoster** [comment]
Yeah I thought the same, but that would mean you cannot mark a draft as private or password protected. 

Still, we can probably move the "Convert to draft" link to that panel, but leave the radios as they are.

**Comment by javierarce** [comment]
Oh, that's a good point. Yes, we could have the option there but not as a radio button (maybe a toggle?)

**Comment by noisysocks** [comment]
**👋 May 18 update**

I've finished updating the `DateTimePicker` component which is used by the Publish / Date popover to use TypeScript (https://github.com/WordPress/gutenberg/pull/40775), React Testing Library (https://github.com/WordPress/gutenberg/pull/40754 https://github.com/WordPress/gutenberg/pull/40957), a newer version of `react-dates` (https://github.com/WordPress/gutenberg/pull/40801), and Storybook controls (https://github.com/WordPress/gutenberg/pull/40986).

With the internals of `DateTimePicker` now in good shape, I'm currently focused on design updates to the Publish / Date popover (https://github.com/WordPress/gutenberg/pull/41097).

Will then continue my way down the list of tasks in the description of this issue.

**Comment by noisysocks** [comment]
Cross posting some design work for a post author dropdown that @javierarce put together in https://github.com/WordPress/gutenberg/pull/41430#issuecomment-1157690629:

> Even though this PR is closed, I'll leave them here as a reference and will share them again with more detail when @ciampo publishes the new exploration:
>
> - [Figma file](https://www.figma.com/file/8vEDwRwmzWhxvcvhc6yaQk/?node-id=1207%3A28788) with the proposal
> - Interactive prototypes:
>   - [Attached flow](https://www.figma.com/proto/8vEDwRwmzWhxvcvhc6yaQk/Status-and-visibility?node-id=1207%3A28788&viewport=-1338%2C179%2C0.46&scaling=min-zoom&starting-point-node-id=1213%3A33643&show-proto-sidebar=1)
>   - [Detached flow](https://www.figma.com/proto/8vEDwRwmzWhxvcvhc6yaQk/Status-and-visibility?node-id=1207%3A28788&viewport=-1338%2C179%2C0.46&scaling=min-zoom&starting-point-node-id=1221%3A60969&show-proto-sidebar=1)

**Comment by noisysocks** [comment]
**Update**

The new designs for all popovers (_Visibility_, _Publish_, _URL_, _Template_) have been implemented and are live in the Gutenberg plugin.

Work is underway on adding new UI for _Featured image_, _Title_ and _Excerpt_ in https://github.com/WordPress/gutenberg/pull/42352.

Work is underway on rehoming the _Mark as pending_, _Revision history_ and _Trash post_ buttons in https://github.com/WordPress/gutenberg/pull/42175.

Implementing the new design for the _Author_ dropdown requires some underlying work on `CustomSelectControl` which @ciampo is exploring in https://github.com/WordPress/gutenberg/pull/41726. If this is too slow for WP 6.1 though then we can resort to the more temporary approach taken in https://github.com/WordPress/gutenberg/pull/41430.

When the above PRs are merged I think we can consider this project complete.

**Comment by javierarce** [comment]
I've been trying to find a good way to solve the problem that @ZebulanStanphill pointed out  about the label + value rows not being clear enough (and maybe even #470):

> It genuinely is confusing to have buttons not be labeled according to their actions. If I see blue text with the name of an author, I might think that it's a link to their profile, rather than a way to edit the author list.

While thinking of a solution I came across [this tweet](https://twitter.com/simonbs/status/1536049033789202435) that shows how the new iOS version is solving a similar problem. 

So in the same spirit, I'd like to get feedback on the following idea: placing an icon next to each value to indicate that the element is not a regular link/button.

The following image shows several ideas for that icon: a double chevron, a horizontal ellipsis, a pencil, sliders, and a cog.

![image](https://user-images.githubusercontent.com/4933/179248698-6aed07c1-0e61-4aa3-bf48-97202b60f890.png)

**Comment by javierarce** [comment]
And here's another idea to group the settings while at the same time making the connection between labels and values clearer.

In the image below the settings are grouped using a border around each item, in a table-like configuration. I've also added a version with an icon on the right side, as I did in my previous comment. 

![image](https://user-images.githubusercontent.com/4933/179557885-4828cefd-232e-407d-9f36-2f22937ca68f.png)

A similar solution was also shared by @shaunandrews in [this post](https://shaunandrews.com/2021/09/the-wordpress-editor-document-status-and-visibility). I think adding icons as he did could also work, but I prefer a purely text-based version since the horizontal space is limited.

**Comment by javierarce** [comment]
After excellent work led by @noisysocks and @Mamaduka, we are quite close to getting most of the original proposal implemented, but I'd like to get final approval from @WordPress/gutenberg-design, @mtias, and other participants on this thread before we go ahead and merge the following changes:

- [Add Featured Image, Title and Excerpt to top of the settings sidebar #42352](https://github.com/WordPress/gutenberg/pull/42352)
- [Add UI for toggling Summary fields #42546](https://github.com/WordPress/gutenberg/pull/42546)

Back on this thread, @jasmussen mentioned that:

> One take-away worth revisiting is the flow of _how_ excerpts and images are added: perhaps instead of being placeholders to fill out, the summary can be mostly a static representation of only that which is added. The act of adding these can then remain a separate part of the summary. Something to explore!

This is something I've explored [in this quick iteration](https://www.figma.com/file/8vEDwRwmzWhxvcvhc6yaQk/Status-%26-visibility?node-id=1563%3A84405), where users can toggle the preview of the Featured Image and Excerpt in the summary, but the redundancy of those elements feels difficult to defend and could be confusing for the user. 

I still think that moving the Featured Image and Excerpt sections to the top of the sidebar makes sense, even if not all users fill them, because it promotes good practices, improves the IA, and serves as an elegant (as Joen said) summary of the publication, but I'd like to hear more voices.

**Comment by critterverse** [comment]
Thanks for the summary @javierarce! It's exciting to see how far this work has come so far.

I'd like to suggest a change to https://github.com/WordPress/gutenberg/pull/42352 because the current implementation shares many of the same issues with the previous exploration that was reverted (see @jasmussen's [summary of that exploration above](https://github.com/WordPress/gutenberg/issues/39082#issuecomment-1091129158)).

Can we instead try this summary panel as a part of the pre-publish panel? That way we won't be putting too much prominence on the Featured Image and Excerpt by having those appear right at the top of the document settings. This would also allow us to keep the existing Featured Image and Excerpt panels without having confusing duplication in the UI.

What do others think about giving this a try?

**Comment by javierarce** [comment]
Thanks, @critterverse! This sounds good to me. The pre-publish panel could benefit from a small redesign, and it's a logical place to show the featured image and excerpt. I'll offer a design for this, open a new issue, and close this one if that's ok.

**Comment by javierarce** [comment]
Ok, I've opened an issue to [enhance the pre-publish sidebar](https://github.com/WordPress/gutenberg/issues/43023), and I'm closing this one.

**Comment by Walrus117** [comment]
All done. I'll post a plugin for it soon. 
<img width="288" alt="Screenshot 2023-10-19 at 20 28 20" src="https://github.com/WordPress/gutenberg/assets/121443452/dbad464a-90ca-4bbd-a1f6-c4c0c7e52220">

**Comment by ghost** [comment]
@WalrusRIK do you have any updates on this plugin? This is exactly what I'm looking for.

**Comment by Walrus117** [comment]
It is complete and I'm using it. Unfortunately I'm not familiar with the Wordpress plugin upload procedures and having taken a look a few weeks ago it appears to be both time consuming and has a delay of several months on their review process. I suppose I could provide it privately, it does represent a week of work however.

**Comment by ghost** [comment]
@WalrusRIK ok, don't do it because of me, but if you ever have something thats available to look at, I'd be very happy to check it out!

**Comment by Walrus117** [comment]
> @WalrusRIK ok, don't do it because of me, but if you ever have something thats available to look at, I'd be very happy to check it out!

I should probably have someone else test it anyway. Would you be open to trying it but not sharing it for now. I've created two versions. A simple panel that gives the ability to control page title and excerpt without needing to add the relevant blocks and the graphical version that looks and functions exactly like the mock up in this post. The idea could be that the simple one is free and the graphical one has a small fee.

**Comment by ivantomicic-crowdhouse** [comment]
> functions

Absolutely!

**Comment by ivantomicic-crowdhouse** [comment]
> > @WalrusRIK ok, don't do it because of me, but if you ever have something thats available to look at, I'd be very happy to check it out!
> 
> I should probably have someone else test it anyway. Would you be open to trying it but not sharing it for now. I've created two versions. A simple panel that gives the ability to control page title and excerpt without needing to add the relevant blocks and the graphical version that looks and functions exactly like the mock up in this post. The idea could be that the simple one is free and the graphical one has a small fee. 

Absolutely!

**Comment by ivantomicic-crowdhouse** [comment]
> > @WalrusRIK ok, don't do it because of me, but if you ever have something thats available to look at, I'd be very happy to check it out!
> 
> I should probably have someone else test it anyway. Would you be open to trying it but not sharing it for now. I've created two versions. A simple panel that gives the ability to control page title and excerpt without needing to add the relevant blocks and the graphical version that looks and functions exactly like the mock up in this post. The idea could be that the simple one is free and the graphical one has a small fee. 

Absolutely! Feel free to ping me at ivan@tomicic.dev

**Comment by ghost** [comment]
> > @WalrusRIK ok, don't do it because of me, but if you ever have something thats available to look at, I'd be very happy to check it out!
> 
> I should probably have someone else test it anyway. Would you be open to trying it but not sharing it for now. I've created two versions. A simple panel that gives the ability to control page title and excerpt without needing to add the relevant blocks and the graphical version that looks and functions exactly like the mock up in this post. The idea could be that the simple one is free and the graphical one has a small fee.

Absolutely! Feel free to ping me at ivan@tomicic.dev

**Comment by Walrus117** [comment]
> > > @WalrusRIK ok, don't do it because of me, but if you ever have something thats available to look at, I'd be very happy to check it out!
> > 
> > 
> > I should probably have someone else test it anyway. Would you be open to trying it but not sharing it for now. I've created two versions. A simple panel that gives the ability to control page title and excerpt without needing to add the relevant blocks and the graphical version that looks and functions exactly like the mock up in this post. The idea could be that the simple one is free and the graphical one has a small fee.
> 

I'll clean it all up a bit and get something over to you. R.
...SENT
